pta-intelligent-search-metadata-extractor/src/main/java/fi/maanmittauslaitos/pta/search/metadata/json/CKANMetadataExtractorConfigurationFactory.java

```diff
@@ -168,42 +171,67 @@ public DocumentProcessingConfiguration createMetadataDocumentProcessingConfigura
-	public DocumentProcessor createMetadataDocumentProcessor() throws ParserConfigurationException {
-		DocumentProcessingConfiguration configuration = createMetadataDocumentProcessingConfiguration();
-		return getDocumentProcessorFactory().createJsonProcessor(configuration);
-	}
+		// CKAN custom field used to extract creation date
+		FieldExtractorConfigurationImpl creationDateExtractor = (FieldExtractorConfigurationImpl) createFirstMatchingJsonPathExtractor(
+				ResultMetadataFields.CKAN_CREATION_DATE,
+				"$.resources[0].created", "$.metadata_created"
+		);
+		creationDateExtractor.setTrimmer(dStamp -> dStamp != null ? dStamp.trim().split("\\.")[0] : null);
+		extractors.add(creationDateExtractor);
 
-	private FieldExtractorConfiguration createJsonPathExtractor(String field, FieldExtractorType type, String jsonPath) {
-		FieldExtractorConfigurationImpl ret = new FieldExtractorConfigurationImpl();
-		ret.setField(field);
-		ret.setType(type);
-		ret.setQuery(jsonPath);
-		return ret;
-	}
 
-	private FieldExtractorConfiguration createCustomJsonPathExtractor(String field, CustomExtractor extractor, String jsonPath) {
-		FieldExtractorConfigurationImpl ret = (FieldExtractorConfigurationImpl) createJsonPathExtractor(field,
-				FieldExtractorType.CUSTOM_CLASS, jsonPath);
-		ret.setCustomExtractor(extractor);
-		return ret;
+		// organizations
+		extractors.add(createCustomListJsonPathExtractor(
+				ResultMetadataFields.ORGANISATIONS_RESOURCE,
+				new ResponsiblePartyCKANCustomExtractor(),
+				"$.['reporting_organization','reporting_person_email']"
+
+		));
+
+		extractors.add(createCustomListJsonPathExtractor(
+				ResultMetadataFields.ORGANISATIONS_METADATA,
+				new ResponsiblePartyCKANCustomExtractor(),
+				"$.['author', 'author_email']"
+
+		));
+
+		// we use a dummy key so the fields are parsed in the same format as the default org config and thus we can use the default
+		// ResponsiblePartyCKANCustomExtractor for this also.
+		extractors.add(createCustomListJsonPathExtractor(
+				ResultMetadataFields.ORGANISATIONS_OTHER,
+				new SimpleResponsiblePartyCKANCustomExtractor(),
+				"$.['a_key_that_should_never_exist_or_else_this_might_break', 'reporting_organization_others']"
+		));
+
+		// Keywords
+		extractors.add(createJsonPathExtractor(
+				ResultMetadataFields.KEYWORDS_ALL,
+				FieldExtractorConfigurationImpl.FieldExtractorType.ALL_MATCHING_VALUES,
+				"$.tag_keyword"
+		));
+
+
+		return configuration;
 	}
 
-	private FieldExtractorConfiguration createCustomListJsonPathExtractor(String field, ListCustomExtractor extractor, String jsonPath) {
-		FieldExtractorConfigurationImpl ret = (FieldExtractorConfigurationImpl) createJsonPathExtractor(field,
-				FieldExtractorType.CUSTOM_CLASS_SINGLE_VALUE, jsonPath);
-		ret.setListCustomExtractor(extractor);
-		return ret;
+	@Override
+	public DocumentProcessor createMetadataDocumentProcessor() throws ParserConfigurationException {
+		return createMetadataDocumentProcessor(null);
```

Comment:
Same pattern as with the ISO metadata config factory

Reply:
Fixed